services/m3dbnode/config/config.go

```diff
@@ -0,0 +1,145 @@
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package config
+
+import (
+	"time"
+
+	etcdclient "github.com/m3db/m3cluster/client/etcd"
+	"github.com/m3db/m3db/client"
+	"github.com/m3db/m3x/instrument"
+	xlog "github.com/m3db/m3x/log"
+)
+
+// Configuration is the configuration for a M3DB node.
+type Configuration struct {
+	// Logging configuration.
+	Logging xlog.Configuration `yaml:"logging"`
+
+	// Metrics configuration.
+	Metrics instrument.MetricsConfiguration `yaml:"metrics"`
+
+	// The host and port on which to listen
+	ListenAddress string `yaml:"listenAddress" validate:"nonzero"`
+
+	// The HTTP host and port on which to listen for the cluster service
+	HTTPClusterListenAddress string `yaml:"httpClusterListenAddress" validate:"nonzero"`
+
+	// The HTTP host and port on which to listen for the node service
+	HTTPNodeListenAddress string `yaml:"httpNodeListenAddress" validate:"nonzero"`
+
+	// The host and port on which to listen for debug endpoints
+	DebugListenAddress string `yaml:"debugListenAddress"`
+
+	// HostID is the local host ID configuration.
+	HostID HostIDConfiguration `yaml:"hostID"`
+
+	// Client configuration, used for inter-node communication and when used as a coordinator.
+	Client client.Configuration `yaml:"client"`
+
+	// The initial garbage collection target percentage
+	GCPercentage int `yaml:"gcPercentage" validate:"max=100"`
+
+	// Write new series asynchronously for fast ingestion of new ID bursts
+	WriteNewSeriesAsync bool `yaml:"writeNewSeriesAsync"`
+
+	// Write new series limit per second to limit overwhelming during new ID bursts
+	WriteNewSeriesLimitPerSecond int `yaml:"writeNewSeriesLimitPerSecond"`
+
+	// Write new series backoff between batches of new series insertions
+	WriteNewSeriesBackoffDuration time.Duration `yaml:"writeNewSeriesBackoffDuration"`
+
+	// Bootstrap configuration
+	Bootstrap BootstrapConfiguration `yaml:"bootstrap"`
+
+	// The filesystem configuration for the node
+	Filesystem FilesystemConfiguration `yaml:"fs"`
+
+	// TickInterval controls the tick interval for the node
+	TickInterval time.Duration `yaml:"tickInterval" validate:"nonzero"`
+
+	// The commit log policy for the node
+	CommitLog commitLogPolicy `yaml:"commitlog"`
+
+	// The repair policy for repairing in-memory data
+	Repair repairPolicy `yaml:"repair"`
+
+	// The pooling policy
+	PoolingPolicy PoolingPolicy `yaml:"poolingPolicy"`
+
+	// The configuration for config service client
+	ConfigService etcdclient.Configuration `yaml:"configService"`
+}
+
+type commitLogPolicy struct {
```

Comment:
Could you make this (& other config structs) exported? Will be useful for when we're overriding properties programmatically, e.g. dtest runs